
# HG changeset patch
# User jcm
# Date 1484137609 28800
# Node ID 8dfbb002197a8e9dfa2881d33ec282fd7a449c25
# Parent  c7140a91e56a846a9691f81c744fd26609de093c
8164293: HotSpot leaking memory in long-running requests
Summary: Applied RMs in sweep_code_cache and related codes.
Reviewed-by: kvn, thartmann

diff -r c7140a91e56a -r 8dfbb002197a src/share/vm/code/nmethod.cpp
--- a/hotspot/src/share/vm/code/nmethod.cpp	Thu Jan 05 18:55:20 2017 -0500
+++ b/hotspot/src/share/vm/code/nmethod.cpp	Wed Jan 11 04:26:49 2017 -0800
@@ -1151,6 +1151,7 @@
 // Clear ICStubs of all compiled ICs
 void nmethod::clear_ic_stubs() {
   assert_locked_or_safepoint(CompiledIC_lock);
+  ResourceMark rm;
   RelocIterator iter(this);
   while(iter.next()) {
     if (iter.type() == relocInfo::virtual_call_type) {
diff -r c7140a91e56a -r 8dfbb002197a src/share/vm/runtime/sweeper.cpp
--- a/hotspot/src/share/vm/runtime/sweeper.cpp	Thu Jan 05 18:55:20 2017 -0500
+++ b/hotspot/src/share/vm/runtime/sweeper.cpp	Wed Jan 11 04:26:49 2017 -0800
@@ -319,6 +319,7 @@
 }
 
 void NMethodSweeper::sweep_code_cache() {
+  ResourceMark rm;
   Ticks sweep_start_counter = Ticks::now();
 
   _flushed_count                = 0;
@@ -626,6 +627,7 @@
 // state of the code cache if it's requested.
 void NMethodSweeper::log_sweep(const char* msg, const char* format, ...) {
   if (PrintMethodFlushing) {
+    ResourceMark rm;
     stringStream s;
     // Dump code cache state into a buffer before locking the tty,
     // because log_state() will use locks causing lock conflicts.
@@ -643,6 +645,7 @@
   }
 
   if (LogCompilation && (xtty != NULL)) {
+    ResourceMark rm;
     stringStream s;
     // Dump code cache state into a buffer before locking the tty,
     // because log_state() will use locks causing lock conflicts.

